processor/blockprocessor/internal/catchupservice.go

```diff
@@ -66,6 +66,7 @@ func CatchupServiceCatchup(ctx context.Context, logger *log.Logger, catchpoint,
 
 	start := time.Now()
 	cfg := config.AutogenLocal
+	cfg.DeadlockDetection = 0
```

Comment:
I don't think this will have any impact.. you need to disable in your indexer main.go or daemon.go at startup time.

Reply:
Startup of what? This the config passed into the service, it doesn't exist until this function.

Reply:
Like this 
```diff
diff --git a/cmd/algorand-indexer/main.go b/cmd/algorand-indexer/main.go
index c1540fc..dff7626 100644
--- a/cmd/algorand-indexer/main.go
+++ b/cmd/algorand-indexer/main.go
@@ -11,6 +11,7 @@ import (
    "github.com/spf13/cobra/doc"
    "github.com/spf13/viper"

+   "github.com/algorand/go-deadlock"
    bg "github.com/algorand/indexer/cmd/block-generator/core"
    v "github.com/algorand/indexer/cmd/validator/core"
    "github.com/algorand/indexer/config"
@@ -94,6 +95,9 @@ func indexerDbFromFlags(opts idb.IndexerDbOptions) (idb.IndexerDb, chan struct{}
 }

 func init() {
+   // disable go-deadlock detection
+   deadlock.Opts.Disable = true
+
    // Utilities subcommand for more convenient access to useful testing utilities.
    utilsCmd := &cobra.Command{
        Use:   "util",
```

Reply:
The go-algorand objects you are passing cfg to don't do anything with cfg.DeadlockDetection — this config value (and/or build flag) is checked and set at service start / init time.

Reply:
Fixed